presto-main/src/main/java/com/facebook/presto/operator/OrderByOperator.java

```diff
@@ -218,4 +295,60 @@ public void close()
         }
         return types.build();
     }
+
+    public WorkProcessor<Page> unspill()
+    {
+        List<WorkProcessor<Page>> spilledPages = getSpilledPages();
+
+        if (spilledPages.isEmpty()) {
+            pageIndex.sort(sortChannels, sortOrder);
+
+            return WorkProcessor.fromIterator(pageIndex.getSortedPages());
+        }
+
+        return mergeSpilledPagesAndCurrentPagesIndex(pageIndex, spilledPages);
+    }
+
+    public List<WorkProcessor<Page>> getSpilledPages()
+    {
+        if (!spiller.isPresent()) {
+            return ImmutableList.of();
+        }
+
+        return spiller.get().getSpills().stream()
+                .map(WorkProcessor::fromIterator)
+                .collect(Collectors.toList());
+    }
+
+    public WorkProcessor<Page> mergeSpilledPagesAndCurrentPagesIndex(PagesIndex currentPagesIndex, List<WorkProcessor<Page>> spilledPages)
+    {
+        PageWithPositionComparator pageWithPositionComparator = new SimplePageWithPositionComparator(sourceTypes, sortChannels, sortOrder);
+        WorkProcessor<Page> pageIndexPages = WorkProcessor.fromIterator(currentPagesIndex.getSortedPages());
+
+        spilledPages.add(pageIndexPages);
+
+        return mergeSortedPages(
+                spilledPages,
+                requireNonNull(pageWithPositionComparator, "comparator is null"),
+                sourceTypes,
+                operatorContext.aggregateUserMemoryContext(),
+                operatorContext.getDriverContext().getYieldSignal());
+    }
+
+    private void updateMemoryUsage(boolean useRevocableMemory)
+    {
+        if (useRevocableMemory) {
+            if (!spillEnabled) {
```

Comment:
`checkState(spillEnabled, "Revocable memory requested when spill to disk is not enabled"); `